lib/WebService/Async/Onfido.pm

```diff
@@ -122,7 +122,7 @@ sub applicant_list {
         }, sub {
             my ($err, @details) = @_;
             $log->errorf('Failed to request document_list: %s', $err);
-            $src->completed->fail($err, @details) unless $src->completed->is_ready;
+            $src->fail($err, @details) unless $src->is_ready;
```

Comment:
Just want to study, now Ryu::Source can be think as a Future ? Or it should be `$f->fail(....)` here ?

Reply:
I am pretty much a blank sheet on this. I think we have made a few changes to `Ryu::Source` and inherited some of the `Future` methods into it. So, `$src->fail` is actually `$src->_completed->fail`

Reply:
Source is [here](https://metacpan.org/dist/Ryu/source/lib/Ryu/Source.pm#L2146)

Reply:
Ok you are right, it is here: https://github.com/regentmarkets/cpan/pull/2641/files#diff-35713b4c2aca4cbbbd087e66447defa8b007e4826321f90f7e769536737a9191L2035

```
for my $k (qw(then fail on_ready transform is_ready is_done is_failed failure else)) {
    do { no strict 'refs'; *$k = $_ } for sub { shift->_completed->$k(@_) }
}
```